src/components/landing/pages/login/auth/via-ldap.tsx

```diff
@@ -1,30 +1,70 @@
-import React, {Fragment} from "react";
-import {Trans} from "react-i18next";
-import {Button, Form} from "react-bootstrap";
+import React, {Fragment, useState} from "react";
+import {Trans, useTranslation} from "react-i18next";
+import {Alert, Button, Form} from "react-bootstrap";
+import {postLdapLogin} from "../../../../../api/user";
+import {getAndSetUser} from "../../../../../utils/apiUtils";
+import {useSelector} from "react-redux";
+import {ApplicationState} from "../../../../../redux";
 
 const ViaLdap: React.FC = () => {
+    const {t} = useTranslation();
+    const ldapCustomName = useSelector((state: ApplicationState) => state.backendConfig.customAuthNames.ldap);
+    const [username, setUsername] = useState("");
+    const [password, setPassword] = useState("");
+    const [error, setError] = useState(false);
+    const login = (event: any) => {
+        postLdapLogin(username, password)
+            .then(loginJson => {
+                console.log(loginJson)
+                getAndSetUser();
+            }).catch(_reason => {
+                setError(true);
+            }
+        )
+        event.preventDefault();
+    }
+
+    const name = ldapCustomName ? `${ldapCustomName} (LDAP)` : "LDAP";
```

Comment:
Why are you adding `(LDAP)` to a custom name?

Reply:
I didn't change it to before https://github.com/codimd/server/blob/master/public/views/shared/signin-modal.ejs#L60

Reply:
okay